pkg/ext/delegate_error_test.go

```diff
@@ -0,0 +1,1109 @@
+				createResult: &TestType{},
+				err:          nil,
+			},
+			wantErr: false,
+		},
+		{
+			name:       "missing user in the context",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {},
+			input: input{
+				ctx:     context.Background(),
+				obj:     &TestType{},
+				options: &metav1.CreateOptions{},
+			},
+			expect: output{
+				createResult: nil,
+				err:          errMissingUserInfo,
+			},
+			wantErr: true,
+		},
+		{
+			name: "store create error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().Create(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, assert.AnError)
+			},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				obj:     &TestType{},
+				options: &metav1.CreateOptions{},
+			},
+			expect: output{
+				createResult: nil,
+				err:          assert.AnError,
+			},
+			wantErr: true,
+		},
+		{
+			name:       "wrong type error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				obj:     &runtime.Unknown{},
+				options: &metav1.CreateOptions{},
+			},
+			expect: output{
+				createResult: nil,
+				err:          assert.AnError,
+			},
+			wantErr: true,
+		},
+		{
+			name:       "create validation error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				obj: &TestType{},
+				createValidation: func(ctx context.Context, obj runtime.Object) error {
+					return assert.AnError
+				},
+				options: &metav1.CreateOptions{},
+			},
+			expect: output{
+				createResult: &TestType{},
+				err:          assert.AnError,
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			addToSchemeTest(scheme)
+
+			gvk := schema.GroupVersionKind{Group: "ext.cattle.io", Version: "v1", Kind: "TestType"}
+			gvr := schema.GroupVersionResource{Group: "ext.cattle.io", Version: "v1", Resource: "testtypes"}
+
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := NewMockStore[*TestType, *TestTypeList](ctrl)
+			tt.storeSetup(mockStore)
+
+			testDelegate := &delegateError[*TestType, *TestTypeList]{
+				inner: &delegate[*TestType, *TestTypeList]{
+					scheme: scheme,
+					t:      &TestType{},
+					tList:  &TestTypeList{},
+					gvk:    gvk,
+					gvr:    gvr,
+					store:  mockStore,
+				},
+			}
+
+			result, err := testDelegate.Create(tt.input.ctx, tt.input.obj, tt.input.createValidation, tt.input.options)
+			if tt.wantErr {
+				// check if we have an error
+				assert.Error(t, err)
+
+				// check if error is an apierror
+				if _, ok := err.(apierrors.APIStatus); ok {
+					assert.True(t, ok)
+					return
+				}
```

Comment:
Should this be

```suggestion
				_, ok := err.(apierrors.APIStatus)
				assert.True(t, ok)
```

? Right now if the error returned is not an APIStatus there just won't be any assert.